Project/Source/Modules/ModulePhysics.cpp

```diff
@@ -0,0 +1,396 @@
+							newPos.y = capsuleCol->GetOwner().GetComponent<ComponentTransform>()->GetGlobalPosition().y;
+							capsuleCol->GetOwner().GetComponent<ComponentTransform>()->SetGlobalPosition(newPos);
+						}
+						break;
+					}
+					default:
+						break;
+					}
+				}
+			}
+		}
+	}
+	return UpdateStatus::CONTINUE;
+}
+
+UpdateStatus ModulePhysics::Update() {
+	if (App->time->IsGameRunning()) {
+		/*if (App->input->GetKey(SDL_SCANCODE_F1) == KS_DOWN) //TODO: DOnt do it by keyboard!!
+		debug = !debug;*/
+
+		if (debug == true) {
+			world->debugDrawWorld();
+		}
+
+		if (App->input->GetKey(SDL_SCANCODE_SPACE) == KS_DOWN) {
+			//btCollisionShape* colShape = new btSphereShape(2.f);
+			btCollisionShape* colShape = new btBoxShape(btVector3(1, 1, 1));
+
+			btVector3 localInertia(0, 0, 0);
+			colShape->calculateLocalInertia(3.0f, localInertia);
+
+			float3 position = App->camera->GetEngineCamera()->frustum.Pos();
+			btDefaultMotionState* myMotionState = new btDefaultMotionState(btTransform(btQuaternion::getIdentity(), btVector3(position.x, position.y, position.z)));
+			btRigidBody::btRigidBodyConstructionInfo rbInfo(3.0f, myMotionState, colShape, localInertia);
+			btRigidBody* body = new btRigidBody(rbInfo);
+
+			world->addRigidBody(body, WorldLayers::WORLD_ELEMENTS, WorldLayers::PLAYER | WorldLayers::WORLD_ELEMENTS | WorldLayers::EVERYTHING);
+			float3 f = App->camera->GetEngineCamera()->frustum.Front();
+			body->applyCentralImpulse(btVector3(f.x * 73.f, f.y * 73.f, f.z * 73.f));
+			/*
+
+
+
+			Sphere bullet;
+			bullet.color = Green;
+			bullet.radius = 1.0f;
+			bullet.SetPos(App->camera->Position.x, App->camera->Position.y, App->camera->Position.z);
+
+			App->physics->AddBody(bullet)->Push(-App->camera->Z.x * 20.0f, -App->camera->Z.y * 20.0f, -App->camera->Z.z * 20.0f);*/
+		}
+	}
+	return UpdateStatus::CONTINUE;
+}
+
+bool ModulePhysics::CleanUp() {
+	// TODO: clean necessary module lists/vectors
+	ClearPhysicBodies();
+
+	RELEASE(world);
+
+	RELEASE(debugDrawer);
+	RELEASE(constraintSolver);
+	RELEASE(broadPhase);
+	RELEASE(dispatcher);
+	RELEASE(collisionConfiguration);
+	// world
+
+	return true;
+}
+
+void ModulePhysics::CreateSphereRigidbody(ComponentSphereCollider* sphereCollider) {
+	sphereCollider->motionState = MotionState(sphereCollider, sphereCollider->centerOffset, sphereCollider->freezeRotation);
+	sphereCollider->rigidBody = AddSphereBody(&sphereCollider->motionState, sphereCollider->radius, sphereCollider->colliderType == ColliderType::DYNAMIC ? sphereCollider->mass : 0);
+	sphereCollider->rigidBody->setUserPointer(sphereCollider);
+	AddBodyToWorld(sphereCollider->rigidBody, sphereCollider->colliderType, sphereCollider->layer);
+}
+
+btRigidBody* ModulePhysics::AddSphereBody(MotionState* myMotionState, float radius, float mass) {
+	btCollisionShape* colShape = new btSphereShape(radius);
+
+	btVector3 localInertia(0, 0, 0);
+	if (mass != 0.f)
+		colShape->calculateLocalInertia(mass, localInertia);
```

Comment:
One-line ifs in separate lines should be avoided. There are a few instances in this file.

Reply:
Ah the ctrl+K+D